tests/components/notify/test_init.py

```diff
@@ -97,3 +127,202 @@ async def test_warn_template(hass, caplog):
     # We should only log it once
     assert caplog.text.count("Passing templates to notify service is deprecated") == 1
     assert hass.states.get("persistent_notification.notification") is not None
+
+
+async def test_invalid_platform(hass, caplog, tmp_path):
+    """Test service setup with an invalid platform."""
+    loaded_platform = MockNotifyPlatform()
+    mock_platform(hass, "testnotify.notify", loaded_platform)
+    integration = hass.data[DATA_INTEGRATIONS]["testnotify"]
+    integration.file_path = tmp_path
+    # Setup the second testnotify2 platform dynamically
+    await async_load_platform(
+        hass,
+        "notify",
+        "testnotify",
+        {"notify": [{"platform": "testnotify"}]},
+        hass_config={"notify": [{"platform": "testnotify"}]},
+    )
+    await hass.async_block_till_done()
+    assert "Invalid notify platform" in caplog.text
```

Comment:
Should we check that this occurs twice as we try to set up two platforms, one from config and one from discovery?